c++/src/kj/compat/http.c++

```diff
@@ -1867,7 +1883,11 @@ public:
     if (length == 0) inner.finishBody();
   }
   ~HttpFixedLengthEntityWriter() noexcept(false) {
-    if (length > 0) inner.abortBody();
+    if (length > 0) {
+      inner.abortBody();
+    } else if (inner.isWriteInProgress()) {
```

Comment:
Since both branches execute equivalent code, would `if (length > 0 || inner.isWriteInProgress)` make more sense?

Reply:
Maybe. In part I wanted to test here that finishBody() actually detects the problem.

Is it worth starting over the CI to fix this?

Reply:
NM, fixed it.